packages/lib/src/vis/shared/Axis.tsx

```diff
@@ -58,12 +59,19 @@ function Axis(props: Props) {
 
   return (
     <>
-      <svg className={styles.axis} style={svgSize} data-type={type}>
+      <svg
+        className={styles.axis}
+        data-type={type}
+        style={{
+          width: isAbscissa ? axisLength : offset,
+          height: isAbscissa ? offset : axisLength,
+        }}
+      >
         <AxisComponent
           scale={scale}
           tickFormat={getTickFormatter(domain, axisLength, scaleType)}
           label={label}
-          labelOffset={type === 'abscissa' ? 28 : 42}
+          labelOffset={offset - (isAbscissa ? 32 : 36)}
```

Comment:
The label offset is now computed from the axis offset, which means that if we adjust axis offsets in the future, the labels won't change position again. Note that the resulting label offsets are now a bit bigger than before (the labels are closer to the edge of the vis area and further from the tick labels).